gammapy/datasets/core.py

```diff
@@ -71,7 +71,9 @@ def stat_sum(self):
         """Total statistic given the current model parameters and priors."""
         prior_stat_sum = 0.0
         if self.models is not None:
-            prior_stat_sum = self.models.parameters.prior_stat_sum()
+            prior_stat_sum = (
+                self.models.parameters.prior_stat_sum()
+            )  # This shouldn't be computed here
```

Comment:
it could if we redefine
`datasets.stat_sum = datasets_stat_sum_likelihood+self.models.parameters.prior_stat_sum()`